torch/distributed/_tensor/random.py

```diff
@@ -173,8 +175,8 @@ def _distribute_region(self, spec: DTensorSpec):
         if self.distribute_region_enabled:
             old_offset = self.get_offset("parallel-rng")
             self._set_pre_op_offset(spec)
-            with torch.random.fork_rng(self._devices):
-                torch.cuda.set_rng_state(self.rng_states["parallel-rng"])
+            with torch.random.fork_rng(self._devices, device_type=self._device_type):
```

Comment:
hmm according to the doc, `fork_rng` does not accept a `device_type` arg, how come this would work?

https://pytorch.org/docs/stable/_modules/torch/random.html#fork_rng